lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -44,58 +48,128 @@ function nodeInBody(node) {
  * Get list of all nodes from the document body.
  *
  * @param {Driver} driver
- * @returns {Array<LH.Artifacts.FontSize.DomNodeWithParent>}
+ * @returns {Promise<Array<LH.Artifacts.FontSize.DomNodeWithParent>>}
  */
-function getAllNodesFromBody(driver) {
-  return driver.getNodesInDocument()
-    .then(nodes => {
-      const nodeMap = new Map();
-      nodes.forEach(node => nodeMap.set(node.nodeId, node));
-      nodes.forEach(node => node.parentNode = nodeMap.get(node.parentId));
-      return nodes.filter(nodeInBody);
-    });
+async function getAllNodesFromBody(driver) {
+  const nodes = /** @type {DomNodeMaybeWithParent[]} */ (await driver.getNodesInDocument());
+  /** @type {Map<number, LH.Artifacts.FontSize.DomNodeMaybeWithParent>} */
+  const nodeMap = new Map();
+  nodes.forEach(node => nodeMap.set(node.nodeId, node));
+  nodes.forEach(node => (node.parentNode = nodeMap.get(node.parentId)));
+  // @ts-ignore - once passing through the filter, it's guaranteed to have a parent
+  return nodes.filter(nodeInBody);
```

Comment:
@brendankenny I was struggling with this in my own projects to. Is there no way to fix this with jsdoc?

Reply:
@patrickhulce shouldn't we remove the ts-ignore if it worked?

Reply:
I did. Maybe Github is being too smart for us and showing the original diff for the old convo? 🤷‍♂️